Documentation/teaching/so2/assign1-kprobe-based-tracer.rst

```diff
@@ -2,7 +2,7 @@
 Assignment 1 - Kprobe based tracer
 ==================================
 
```

Comment:
Please use some prefix tags:

e.g Documentation: assign1: Fix 1-tracer documentation

Add small details about what was fixed:
- update the deadline for 2023
- fix style issues
- update links.